File: Course-Registration-System-main/java-(iteration_1)/src/Grade.java

    //This class represents the general information and grades of the course taken by the student.
    
    public class Grade {

    //Codes identifying course and grade information
    private String courseName;
    private String courseCode;
    private int semester;
    private double credit;
    private double numericGrade;
    private String letterGrade;
    
    //Constructor method for the Grade class.
    //Initializes the class properties with the provided values.
    public Grade(String courseName, String courseCode,int semester, double credit, double numericGrade, String letterGrade) {
        this.courseName = courseName;
        this.courseCode = courseCode;
        this.semester=semester;
        this.credit =  credit;
        this.numericGrade = numericGrade;
        this.letterGrade = letterGrade;
    }

    //Getter and setter methods for each property.
    public String getCourseName() {
        return courseName;
    }

    public void setCourseName(String courseName) {
        this.courseName = courseName;
    }

    public String getCourseCode() {
        return courseCode;
    }

    public void setCourseCode(String courseCode) {
        this.courseCode = courseCode;
    }

    public int getSemester() {
        return semester;
    }

    public void setSemester(int semester) {
        this.semester = semester;
    }

    public double getCredit() {
        return credit;
    }

    public void setCredit(double credit) {
        this.credit = credit;
    }

    public double getNumericGrade() {
        return numericGrade;
    }

    public void setNumericGrade(double numericGrade) {
        this.numericGrade = numericGrade;
    }

    public String getLetterGrade() {
        return letterGrade;
    }

    public void setLetterGrade(String letterGrade) {
        this.letterGrade = letterGrade;
    }
}

File: Course-Registration-System-main/python-(iteration_2)/src/Grade.py
class Grade:
    def __init__(self, courseName, courseCode, semester, credit, numericGrade, letterGrade):
        self.__courseName = courseName
        self.__courseCode = courseCode
        self.__semester = semester
        self.__credit = credit
        self.__numericGrade = numericGrade
        self.__letterGrade = letterGrade

    # Getter and setter methods
    def getCourseName(self):
        return self.__courseName
        
    def setCourseName(self, courseName):
        self.__courseName = courseName

    def getCourseCode(self):
        return self.__courseCode

    def setCourseCode(self, courseCode):
        self.__courseCode = courseCode
 
    def getSemester(self):
        return self.__semester

    def setSemester(self, semester):
        self.__semester = semester

    def getCredit(self):
        return self.__credit

    def setCredit(self, credit):
        self.__credit = credit

    def getNumericGrade(self):
        return self.__numericGrade

    def setNumericGrade(self, numericGrade):
        self.__numericGrade = numericGrade

    def getLetterGrade(self):
        return self.__letterGrade

    def setLetterGrade(self, letterGrade):
        self.__letterGrade = letterGrade
